swift_browser_ui/ui/api.py

```diff
@@ -833,3 +833,54 @@ async def get_crypted_upload_session(
             "wssignature": ws_signature,
         }
     )
+
+
+async def get_project_key(request: aiohttp.web.Request) -> aiohttp.web.Response:
+    """Answer project specific encryption keys."""
+    # vault_client: VaultClient = request.app[VAULT_CLIENT]
+    project = request.match_info["project"]
+    # public_key = await vault_client.get_public_key(project)
+
+    upload_endpoint = setd["upload_external_endpoint"]
+
+    # return aiohttp.web.Response(
+    #     text=public_key,
+    # )
+    return aiohttp.web.Response(
+        status=303,
+        headers={
+            "Location": f"{upload_endpoint}/cryptic/{project}/key",
+        },
+    )
+
+
+async def get_object_header(request: aiohttp.web.Request) -> aiohttp.web.Response:
+    """Answer project specific encryption keys."""
+
+    session = await aiohttp_session.get_session(request)
+    request.app["Log"].info(
+        "API call for object header from "
+        f"{request.remote}, sess: {session} :: {time.ctime()}"
+    )
+    project = request.match_info["project"]
+    container = request.match_info["container"]
+    obj = request.match_info["object"]
+
+    upload_endpoint = setd["upload_external_endpoint"]
+    #
+    # runner_id = await open_upload_runner_session(request)
+    # # path = f"/{request.match_info['project']}/{request.match_info['container']}"
+    path = f"{upload_endpoint}/cryptic/{project}/{container}/{obj}/header"
+    # signature = await sign(3600, path)
+    # path += (
+    #     f"?session={runner_id}"
+    #     + f"&signature={signature['signature']}"
+    #     + f"&valid={signature['valid']}"
+    # )
+
+    return aiohttp.web.Response(
+        status=303,
+        headers={
+            "Location": path,
+        },
+    )
```

Comment:
No need to sign the upload API calls, frontend code can ask the UI to create a pre-signed request for a given path whenever required.